File: Summerland_frontend/src/app/houses/house-item/house-details/house-calendar/reservation.ts
export class Reservation{
  private _id:string;
  private _confirmation:string;
  private _date_debut:Date;
  private _date_fin:Date;
  private _customer;
  private _house;


  get customer() {
    return this._customer;
  }

  set customer(value) {
    this._customer = value;
  }

  get house() {
    return this._house;
  }

  set house(value) {
    this._house = value;
  }

  get id(): string {
    return this._id;
  }

  set id(value: string) {
    this._id = value;
  }

  get confirmation(): string {
    return this._confirmation;
  }

  set confirmation(value: string) {
    this._confirmation = value;
  }

  get date_debut(): Date {
    return this._date_debut;
  }

  set date_debut(value: Date) {
    this._date_debut = value;
  }

  get date_fin(): Date {
    return this._date_fin;
  }

  set date_fin(value: Date) {
    this._date_fin = value;
  }
}

File: summerland_backend/src/main/java/com/summerland/entities/Reservation.java
package com.summerland.entities;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.JoinColumn;
import javax.persistence.ManyToOne;
import javax.persistence.Table;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import java.util.Date;


@Entity
@Table(name = "reservations")
public class Reservation {
	
	@Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private long id;
	private String confirmation;
	private Date date_debut;
	private Date date_fin;

	@ManyToOne
	@JoinColumn(name="CUSTOMER")
	private Customer customer;
	

	@ManyToOne
	@JoinColumn(name="HOUSE")

	private House house;

	public Reservation() {
	}

	public Reservation(String confirmation, Date date_debut, Date date_fin, Customer customer, House house) {
		this.confirmation = confirmation;
		this.date_debut = date_debut;
		this.date_fin = date_fin;
		this.customer = customer;
		this.house = house;
	}

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public String getConfirmation() {
		return confirmation;
	}

	public void setConfirmation(String confirmation) {
		this.confirmation = confirmation;
	}

	public Customer getCustomer() {
		return customer;
	}

	public void setCustomer(Customer customer) {
		this.customer = customer;
	}


	public House getHouse() {
		return house;
	}


	public void setHouse(House house) {
		this.house = house;
	}


	public Date getDate_debut() {
		return date_debut;
	}

	public void setDate_debut(Date date_debut) {
		this.date_debut = date_debut;
	}

	public Date getDate_fin() {
		return date_fin;
	}

	public void setDate_fin(Date date_fin) {
		this.date_fin = date_fin;
	}

	@Override
	public String toString() {
		return "Reservation{" +
				"id=" + id +
				", confirmation='" + confirmation + '\'' +
				", date_debut=" + date_debut +
				", date_fin=" + date_fin +
				", customer=" + customer +
				", house=" + house +
				'}';
	}
}
